File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> map1;
        unordered_map<char, char> map2;
        int n = s.size();
        for(int i=0; i<n; i++){
            char ch1 = s[i];
            char ch2 = t[i];
            if((map1.find(ch1) != map1.end() && map1[ch1] != ch2) || (map2.find(ch2) != map2.end() && map2[ch2] != ch1)){
                return false;
            }
            map1[ch1] = ch2;
            map2[ch2] = ch1;
        }
        return true;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character, Character> map1 = new HashMap<>();
        HashMap<Character, Character> map2 = new HashMap<>();
        int n = s.length();
        for(int i=0; i<n; i++){
            char ch1 = s.charAt(i);
            char ch2 = t.charAt(i);
            if((map1.containsKey(ch1) && map1.get(ch1) != ch2) || (map2.containsKey(ch2) && map2.get(ch2) != ch1)){
                return false;
            }
            map1.put(ch1, ch2);
            map2.put(ch2, ch1);
        }
        return true;
    }
}